frontend/components/Timer.js

```diff
@@ -1,74 +1,160 @@
+        sock = new SockJS(url)
+        stompClient = over(sock)
+        stompClient.connect({}, onConnected, onError)
+    }, [])
+
+
+    const onConnected = () => {
+        console.log("CONNECTED")
+        stompClient.subscribe("/response/timer", onMessageReceived)
+        stompClient.send("/timer", {}, JSON.stringify({
+            action: "GET"
+        }));
+    }
+
+    const onError = (error) => {
+        console.log("Error: " + error)
+    }
+
+    const onMessageReceived = (payload) => {
+        let millis = JSON.parse(payload.body).time
+        let minutes = Math.floor(millis / 60000)
+        let seconds = Number(((millis % 60000) / 1000).toFixed(0))
+        setMinutes(minutes)
+        setSeconds(seconds)
+    }
+
     const toggle = () => {
+        if(!isActive) {
+            sendStart()
+        } else {
+            sendStop()
+        }
         setIsActive(!isActive)
     }
 
     const reset = () => {
-        sendTimeToDatabase('-end')
-        setMinutes(0)
-        setSeconds(0)
-        setIsActive(false)
+        if(isActive){
+            setIsActive(!isActive)
+        }
+        sendReset()
+        return Alert.alert(
+            "Are your sure?",
+            "Are you sure you want to reset the timer?",
+            [
+                {
+                    text: "Yes",
+                    onPress: () => {
+                        sendReset()
+                    },
+                },
+                {
+                    text: "No",
+                },
+            ]
+        );
     }
 
-    const sendTimeToDatabase = (param) => {
-        Api.post(`/timer${param}/token`, {
-            minutes: minutes,
-            seconds: seconds
-        }).then(res => {
-            console.log(res)
-        }).catch(e => {
-            console.log(e)
-        })
+    const end = () => {
+        sendEnd()
+        return Alert.alert(
+            "Are your sure?",
+            "Are you sure you want to end the match?",
+            [
+                {
+                    text: "Yes",
+                    onPress: () => {
+                        sendEnd()
+                    },
+                },
+                {
+                    text: "No",
+                },
+            ]
+        );
     }
 
-    useEffect(() => {
-        let interval = null
-        if(isActive){
-            interval = setInterval(() => {
-                if(seconds === 59){
-                    setMinutes(minutes + 1)
-                    setSeconds(0)
-                } else{
-                    setSeconds(seconds + 1)
-                }
-                sendTimeToDatabase('')
-            }, 1000)
-        } else if (!isActive && (seconds !== 0 && minutes !== 0)){
-            clearInterval(interval)
+    const sendStart = () => {
+        if(stompClient) {
+            let message = {
+                action: "START"
+            };
+            stompClient.send('/timer', {}, JSON.stringify(message));
+        }
+    }
```

Comment:
All functions sending actions are similar, so we can change all of them to one with parameter action, but it's working now, so that's only my suggestion. If you want to leave it in that form, I'm approving it.